crates/generator/src/generator.rs

```diff
@@ -698,20 +698,24 @@ impl Generator {
         max_cycles: u64,
         cycles_pool: Option<&mut CyclesPool>,
     ) -> Result<RunResult, TransactionError> {
-        let run_result = self.unchecked_execute_transaction(
+        let args = self.prepare_machine_run_args(
             chain,
             state,
             block_info,
             raw_tx,
             max_cycles,
             cycles_pool,
         )?;
-        Ok(run_result)
+
+        let run_result = self.machine_run(args)?;
+        self.handle_run_result(state, block_info, raw_tx, run_result)
     }
 
-    /// execute a layer2 tx, doesn't check exit code
+    /// execute a layer2 tx without handle run result
+    /// for custom godwoken-scripts contract tests only
+    #[cfg(feature = "unhandled-execution")]
```

Comment:
Seems a comment "tests of godwoken-scripts use only" is enough, how about omitting the "unhandled-execution" feature?